tests/callbacks_test.go

```diff
@@ -601,3 +601,174 @@ func (s *CallbacksSuite) TestNexusResetWorkflowWithCallback() {
 		100*time.Millisecond,
 	)
 }
+
+func blockingWorkflow(ctx workflow.Context) error {
+	return workflow.Await(ctx, func() bool {
+		info := workflow.GetInfo(ctx)
+		return info.OriginalRunID != info.WorkflowExecution.RunID
```

Comment:
I would either just terminate the workflow and check that the callback deliver the failure or explain in a comment here why we are waiting for the run ID to change.